File: Easy/Roman_to_Integer/Roman.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        data = {
            'I': 1,
            'V': 5,
            'X': 10,
            'L': 50,
            'C': 100,
            'D': 500,
            'M': 1000,}

        result, idx  = 0, 0

        while (idx < len(s)):
            next_idx = idx + 1
            a = data.get(s[idx])

            if next_idx < len(s):
                b = data.get(s[next_idx])
                if a >= b:
                    result += a 
                    idx += 1 
                else:
                    result += b - a 
                    idx += 2
            else:
                result += a 
                idx += 1

        return result
            

File: Easy/Roman_to_Integer/Roman.go
package main

func romanToInt(s string) int {
    romans := map[string]int{
            "I": 1,
            "V": 5,
            "X": 10,
            "L": 50,
            "C": 100,
            "D": 500,
            "M": 1000,}
    i := 0 
    result := 0
    len_str := len([]rune(s))

    for i < len_str{
        value := romans[string(s[i])] 
        if i + 1 < len_str{
            nextVal := romans[string(s[i + 1])]
            if value < nextVal{
                result += nextVal - value
                i += 2
            }else{
                result += value 
                i++
            }
        } else{
            result += value
            i++
        }
    }

    return result
}